xds/internal/balancer/edsbalancer/eds_impl.go

```diff
@@ -104,6 +106,13 @@ type edsBalancerImpl struct {
 	innerState             balancer.State // The state of the picker without drop support.
 	serviceRequestsCounter *client.ServiceRequestsCounter
 	serviceRequestCountMax uint32
+
+	// clusterName is the cluster name used for watching EDS. It's attached to
+	// the addresses, and will be read by the google default creds handshaker to
+	// decide to use TLS or ALTS.
+	//
+	// It's a string. Only access atomically.
+	clusterName atomic.Value
```

Comment:
Same here. Unless we have a very valid reason to use an atomic, I wouldn't prefer doing that.

Reply:
Done